mllib/src/main/scala/org/apache/spark/mllib/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -183,11 +183,16 @@ private[spark] object DecisionTreeMetadata extends Logging {
         }
       case _ => featureSubsetStrategy
     }
+
+    val isIntRegex = "^([1-9]\\d*)$".r
+    val isFractionRegex = "^(0\\.\\d*[1-9]\\d*\\d*|1\\.0+)$".r
     val numFeaturesPerNode: Int = _featureSubsetStrategy match {
       case "all" => numFeatures
       case "sqrt" => math.sqrt(numFeatures).ceil.toInt
       case "log2" => math.max(1, (math.log(numFeatures) / math.log(2)).ceil.toInt)
       case "onethird" => (numFeatures / 3.0).ceil.toInt
+      case isIntRegex(number) => if (number.toInt > numFeatures) numFeatures else number.toInt
```

Comment:
This cast will fail if the string integer is greater than `Integer.MAX_VALUE`. Can you add a check for this before doing the cast?